lib/awful/hotkeys_popup/widget.lua

```diff
@@ -100,55 +160,77 @@ function widget.new()
             ['#21']="=",
             Control="Ctrl"
         },
-        _cached_wiboxes = {}
+        _additional_hotkeys = {},
+        _cached_wiboxes = {},
+        _cached_awful_keys = nil,
+        _colors_counter = {},
+        _group_list = {},
+        _widget_settings_loaded = false,
     }
 
-    local cached_awful_keys = nil
-    local colors_counter = {}
-    local colors = beautiful.xresources.get_current_theme()
-    local group_list = {}
+
+    function widget_instance:_load_widget_settings()
+        if self._widget_settings_loaded then return end
+        self.width = args.width or dpi(1200)
+        self.height = args.height or dpi(800)
+        self.bg = args.bg or
+            beautiful.hotkeys_bg or beautiful.bg_normal
+        self.fg = args.fg or
+            beautiful.hotkeys_fg or beautiful.fg_normal
+        self.border_width = args.border_width or
+            beautiful.hotkeys_border_width or beautiful.border_width
+        self.border_color = args.border_color or
+            beautiful.hotkeys_border_color or self.fg
+        self.shape = args.shape or beautiful.hotkeys_shape
+        self.modifiers_fg = args.modifiers_fg or
+            beautiful.hotkeys_modifiers_fg or beautiful.bg_minimize or "#555555"
+        self.label_fg = args.label_fg or
+            beautiful.hotkeys_label_fg or self.bg
+        self.opacity = beautiful.notification_opacity or 1
```

Comment:
As a couple of people already told us on IRC, this still make little sense. Please add a proper variable for it. You can fallback to `beautiful.notification_opacity` is you thing it's worth it. I would not mind the behavior break if this was changed.